modules/in_band_dtmf/in_band_dtmf.c

```diff
@@ -0,0 +1,323 @@
+/**
+ * @file in_band_dtmf.c  DTMF decoder
+ */
+#include <time.h>
+#include <ctype.h>
+#include <string.h>
+#include <re.h>
+#include <rem.h>
+#include <baresip.h>
+
+
+/**
+ * @defgroup in_band_dtmf in_band_dtmf
+ *
+ * Audio filter that de- and encodes DTMF tones.
+ *
+ * The tone length of an encoded tone is 0.1s.
+ * New tones can be added while encoding is active.
+ */
+
+struct in_band_dtmf_filt_dec {
+	struct aufilt_dec_st af;  /* inheritance */
+	struct dtmf_dec *dec;
+	const struct audio *au;
+	struct le le_priv;
+	struct tmr tmr_dtmf_end;
+};
+
+
+struct in_band_dtmf_filt_enc {
+	struct aufilt_enc_st af;  /* inheritance */
+	struct mbuf *mb;
+	unsigned srate;
+	struct le le_priv;
+};
+
+
+static struct list encs;
+static struct list decs;
+
+
+static void dtmfend_handler(void *arg)
+{
+	char digit = (char)(uintptr_t)arg;
+	struct in_band_dtmf_filt_dec *st;
+
+	if (list_isempty(&decs)) {
+		warning("in_band_dtmf: no active call\n");
+		return;
+	}
+
+	st = decs.head->data;
+
+	audio_call_telev_handler(st->au, digit, true);
+}
+
+
+static void in_band_dtmf_dec_handler(char digit, void *arg)
+{
+	(void)arg;
+	struct in_band_dtmf_filt_dec *st;
```

Comment:
Please get the state from the `void *arg` variable.

Then the decs list is not needed anymore.

Reply:
How to manage the memory then? Should `decs` become a `dec` or is there a way to avoid the global variable? 

```
 static struct list encs;
-static struct list decs;
+static struct dec;
```

Reply:
fixed by https://github.com/baresip/baresip/commit/72d9c297ce45b8bfece361889761fd05919be46b